rfcs/rfc0018.md

```diff
@@ -0,0 +1,323 @@
+logic, prototype bypassing, subroutines) and adding another one seemed
+unnecessary.
+
+`?->` was the runner up in the poll, and its popularity is justified: it
+alludes to a ternary check and, even in regexes, to only proceeding if
+whatever came before is there. It also leaves the arrow intact, indicating
+it is checking something that comes _before_ the dereferencing takes place
+(unlike, for example, `->?` or `->>`). It is also the chosen notation for
+most languages that implement this feature, so why surprise developers with
+another way to express what they are already familiar with?
+
+Finally, `//->` was considered but `//` is defined-or, not defined-and as
+this new operator, so it could be even more confusing to developers.
+
+* Why add this just to arrow dereferencing and not to "X"
+
+Because the optional chain is trying to solve the specific (and real-world)
+issue of having to add a lot of repetitive and error-prone boilerplate tests
+on an entire chain individually, and nothing else.
+
+While we could (hypothetically) try to expand this notion to other operators
+(e.g. `?=`, `?=~`, etc), the benefit of doing so is unclear at this point,
+and would require a lot of effort picking appart which operators should
+and shouldn't include the "optional" variation [4].
+
+* Why definedness and not truthfulness? Or 'ref'? Or 'isa'?
+
+Semantically, `undef` means there is nothing there. We still want the code
+to fail loudly if we are dereferencing anything defined, as it would indicate
+something wrong with the code, the underlying logic, or both. It is also
+important to note that Perl allows you to call methods on strings (then
+treated as package/class names), so we cannot reliably test for 'ref'
+without making things really convoluted with assumptions, rules and
+exceptions for each type of data.
+
+* Can we have implicit `?->` after its first use? Have it flip/flop against
+`->`?
+
+The idea is for `?->` to always be explicit, since it is testing the
+lefhand side of the expression. Making the operator change how explicit or
+implicit dereference works in the rest of the expression could lead to
+confusion and bugs.
+
+* Why not other identity values according to context? Like '' when
+concatenating strings, 0 on addition/subtraction and 1 on
+multiplication/division?
+
+While tempting, it would not only produce values that could be confused with
+an actual successful call, it would also mean we'd have to check all types
+of data and agree on their identity value (what is the identity of a sub?
+Or a glob?).
+
+Instead, one could just use the already provided `//` operator to achieve
+the same results, e.g.: `"value is: " . ($ref?->{val} // '')`.
+
+## Open Issues
+
+Optional chains under string interpolation and regular expressions could be
+enabled in the future, hidden behind a feature flag to prevent backwards
+compatibility issues, much like what was done with `postderef_qq`.
```

Comment:
However, a lot of that issue will be largely negated if we obtain Template Literals (See PR #19)

Reply:
I believe optional chains will hardly be necessary/used in regexes and string interpolation, almost by definition, but I will add a nod to template literals on that section either way.